src/core/provider/MCAccountValidator.cpp

```diff
@@ -0,0 +1,412 @@
+            test();
+        }
+    }
+    else if (mCurrentServiceTested == SERVICE_SMTP){
+        if (mCurrentServiceIndex < mSmtpServices->count()) {
+            SMTPAsyncSession *smtpSession = new SMTPAsyncSession();
+            smtpSession->setUsername(mUsername);
+            smtpSession->setPassword(mPassword);
+            
+            mSmtpServer = (NetService *) mSmtpServices->objectAtIndex(mCurrentServiceIndex);
+            smtpSession->setHostname(mSmtpServer->hostname());
+            smtpSession->setPort(mSmtpServer->port());
+            smtpSession->setConnectionType(mSmtpServer->connectionType());
+            
+            mOperation =  (SMTPOperation *)smtpSession->checkAccountOperation(Address::addressWithMailbox(mEmail));
+            mOperation->setCallback(this);
+            mOperation->start();
+
+        }
+        else {
+           callback()->operationFinished(this);
+        }
+    }
+    else {
+        callback()->operationFinished(this);
+    }
+}
+
+void AccountValidator::setEmail(String * email)
+{
+    MC_SAFE_REPLACE_COPY(String, mEmail, email);
+}
+
+String * AccountValidator::email()
+{
+    return mEmail;
+}
+
+void AccountValidator::setUsername(String * username)
+{
+    MC_SAFE_REPLACE_COPY(String, mUsername, username);
+}
+
+String * AccountValidator::username()
+{
+    return mUsername;
+}
+
+void AccountValidator::setPassword(String * password)
+{
+    MC_SAFE_REPLACE_COPY(String, mPassword, password);
+}
+
+String * AccountValidator::password()
+{
+    return mPassword;
+}
+
+void AccountValidator::setOAuth2Token(String * OAuth2Token)
+{
+    MC_SAFE_REPLACE_COPY(String, mOAuth2Token, OAuth2Token);
+}
+
+String * AccountValidator::OAuth2Token()
+{
+    return mOAuth2Token;
+}
+
+void AccountValidator::setImapServices(Array * imapServices)
+{
+    MC_SAFE_REPLACE_COPY(Array, mImapServices, imapServices);
+}
+
+Array * AccountValidator::imapServices()
+{
+    return mImapServices;
+}
+
+void AccountValidator::setSmtpServices(Array * smtpServices)
+{
+    MC_SAFE_REPLACE_COPY(Array, mSmtpServices, smtpServices);
+}
+
+Array * AccountValidator::smtpServices()
+{
+    return mSmtpServices;
+}
+
+void AccountValidator::setPopServices(Array * popServices)
+{
+    MC_SAFE_REPLACE_COPY(Array, mPopServices, popServices);
+}
+
+Array * AccountValidator::popServices()
+{
+    return mPopServices;
+}
+
+String * AccountValidator::identifier()
+{
+    return mIdentifier;
+}
+
+NetService * AccountValidator::imapServer()
+{
+    return mImapServer;
+}
+
+NetService * AccountValidator::smtpServer()
+{
+    return mSmtpServer;
+}
+
+NetService * AccountValidator::popServer()
+{
+    return mPopServer;
+}
+
+ErrorCode AccountValidator::imapError()
+{
+    return mImapError;
+}
+
+ErrorCode AccountValidator::popError()
+{
+    return mPopError;
+}
+
+ErrorCode AccountValidator::smtpError()
+{
+    return mSmtpError;
+}
+
+void POPAsyncSession::setOperationQueueCallback(OperationQueueCallback * callback)
```

Comment:
You probably meant AccountValidator:: setOperationQueueCallback().
Anyway remove those methods.